### ISSUE TITLE
[BUG](?) Strict null check not warning about possibly undefined class members

### ISSUE THREAD

**Issue by Bnaya** [Microsoft/TypeScript#14926]
**TypeScript Version:**  2.2.1

**Code**

```ts
class MyClass {
    public request: {} | null;
}
const c = new MyClass();
 // with strict null check the type here is {} | null, but its actually undefined! 
c.request

```

**Expected behavior:**
The type-checker should warn that request is undefined if no value was set in the member declaration or in the constructor  
**Actual behavior:**
TS treating the member as the declared type, even its not initialize

**Comment by cevek** [comment]
Heh, another one issue https://github.com/Microsoft/TypeScript/issues/14929

**Comment by DanielRosenwasser** [comment]
Duplicate of #8476 which is very difficult to solve.

**Comment by gdelmas** [comment]
> Duplicate of #8476 which is very difficult to solve.

"Very difficult to solve" should not have been the reason to close #8476

**Comment by Bnaya** [comment]
any comments should go to #8476
